src/twisted/internet/unix.py

```diff
@@ -171,28 +172,56 @@ def doRead(self):
             else:
                 return main.CONNECTION_LOST
 
-        if ancillary:
-            ancillaryData = ancillary[0][2]
-            fdCount = len(ancillaryData) // 4
-            fds = struct.unpack('i'*fdCount, ancillaryData)
-            if interfaces.IFileDescriptorReceiver.providedBy(self.protocol):
-                for fd in fds:
-                    self.protocol.fileDescriptorReceived(fd)
+        for cmsgLevel, cmsgType, cmsgData in ancillary:
+            if (cmsgLevel == socket.SOL_SOCKET and
+                cmsgType == sendmsg.SCM_RIGHTS):
+                self._ancillaryLevelSOLSOCKETTypeSCMRIGHTS(cmsgData)
             else:
                 log.msg(
                     format=(
-                        "%(protocolName)s (on %(hostAddress)r) does not "
-                        "provide IFileDescriptorReceiver; closing file "
-                        "descriptor received (from %(peerAddress)r)."),
+                        "%(protocolName)s (on %(hostAddress)r) "
+                        "received unsupported ancillary data "
+                        "(level=%(cmsgLevel)r, type=%(cmsgType)r) "
+                        "from %(peerAddress)r."),
                     hostAddress=self.getHost(), peerAddress=self.getPeer(),
                     protocolName=self._getLogPrefix(self.protocol),
+                    cmsgLevel=cmsgLevel, cmsgType=cmsgType,
                     )
-                for fd in fds:
-                    os.close(fd)
 
         return self._dataReceived(data)
 
 
+    def _ancillaryLevelSOLSOCKETTypeSCMRIGHTS(self, cmsgData):
+
```

Comment:
Ideally this newline should be removed; it's unnecessary, per the coding standard.